File: server/src/main/kotlin/com/blueoauld/server/activity/application/response/ReviewRowResponse.kt
package com.blueoauld.server.activity.application.response

import com.blueoauld.server.activity.repository.result.ReviewResult
import java.time.Instant

data class ReviewRowResponse(

    val reviewId: Long,
    val fromId: Long,
    val toId: Long,
    val nickname: String,
    val content: String,
    val createdAt: Instant
) {

    companion object {
        fun from(result: ReviewResult): ReviewRowResponse {
            return ReviewRowResponse(
                reviewId = result.reviewId,
                fromId = result.fromId,
                toId = result.toId,
                nickname = result.nickname,
                content = result.content,
                createdAt = result.createdAt
            )
        }
    }
}

File: app/app/Model/Activity/Response/ReviewRowResponse.swift
struct ReviewRowResponse: Decodable, Identifiable {
    
    let reviewId: Int64
    let fromId: Int64
    let toId: Int64
    let nickname: String
    let content: String
    let createdAt: String
    
    var id: Int64 { reviewId }
    
    init(from response: ReviewCreateResponse) {
        self.reviewId = response.reviewId
        self.fromId = response.fromId
        self.toId = response.toId
        self.nickname = response.nickname
        self.content = response.content
        self.createdAt = response.createdAt
    }
}
